lib/algolia/index.rb

```diff
@@ -264,6 +264,72 @@ def get_objects(objectIDs, attributes_to_retrieve = nil, request_options = {})
       client.post(Protocol.objects_uri, { :requests => requests }.to_json, :read, request_options)['results']
     end
 
+    #
+    # Find object by the given condition.
+    #
+    # Options can be passed in request_options body:
+    #  - query (string): pass a query
+    #  - paginate (bool): choose if you want to iterate through all the
+    # documents (true) or only the first page (false). Default is true.
+    # The function takes a block to filter the results from search query
+    # Usage example:
+    #  index.find_object({'query' => '', 'paginate' => true}) {|obj| obj.key?('company') and obj['company'] == 'Apple'}
+    #
+    # @param request_options contains extra parameters to send with your query
+    #
+    def find_object(request_options = {})
+      query = ''
+      paginate = true
+      page = 0
```

Comment:
The 2 first initializations are useless since you override it just below. If both `request_options[:query] || request_options['query']` are nil, `query` will be nil and not `''`. Is that what you expected?

Not sure how `search(nil)` is handle behind, it might be fine?

Worst case, you could do `query = request_options[:query] || request_options['query'] || ''`